### ISSUE TITLE
Route versioning causing `DeleteV2` to call the v1 API

### ISSUE THREAD

**Issue by ncw** [dropbox/dropbox-sdk-go-unofficial#49]
Calling `DeleteV2` on a directory causes an error `unexpected end of JSON input` to be returned instead of the actual error.

Tested with dd3c8f54b43f63a7aaf07486f797a14e549aa2c8 (head of master as of now).

Request

```
2018/12/22 16:12:49 DEBUG : HTTP REQUEST (req 0xc000106300)
2018/12/22 16:12:49 DEBUG : POST /2/files/delete HTTP/1.1
Host: api.dropboxapi.com
User-Agent: rclone/v1.45-DEV
Content-Length: 48
Authorization: XXXX
Content-Type: application/json
Accept-Encoding: gzip

{"path":"/rclone-test-sotadic5copuwoz8tefavop0"}
```

Response

```
2018/12/22 16:12:49 DEBUG : HTTP RESPONSE (req 0xc000106300)
2018/12/22 16:12:49 DEBUG : HTTP/1.1 200 OK
Transfer-Encoding: chunked
Cache-Control: no-cache
Connection: keep-alive
Content-Type: application/json
Date: Sat, 22 Dec 2018 16:12:49 GMT
Pragma: no-cache
Server: nginx
Vary: Accept-Encoding
X-Content-Type-Options: nosniff
X-Dropbox-Request-Id: 627c3408ccbe2562f597fea7cad99959
X-Frame-Options: SAMEORIGIN
X-Server-Response-Time: 428

d5
{".tag": "folder", "name": "rclone-test-sotadic5copuwoz8tefavop0", "path_lower": "/rclone-test-sotadic5copuwoz8tefavop0", "path_display": "/rclone-test-sotadic5copuwoz8tefavop0", "id": "id:q
qS0afUHwS0AAAAAAAHDUQ"}
0
```

Which produces the error `unexpected end of JSON input`

This is quite clearly calling the v1 API - it should be calling `delete_v2`.

Using git bisect I found that the offending commit is 118eb0115111626cc1679ca95b8367b9f13653af

```
$ git bisect bad
118eb0115111626cc1679ca95b8367b9f13653af is the first bad commit
commit 118eb0115111626cc1679ca95b8367b9f13653af
Author: Diwaker Gupta <diwaker@dropbox.com>
Date:   Wed Oct 10 22:30:33 2018 -0700

    Add support for route versioning.
    
    Also update API spec. Bumping revision to 5.0.0 -- I'm not aware of any
    breaking changes, but bumping major version anyways to raise awareness
    about route versioning.

:040000 040000 fa13db850e9ca26f417a2953908505709fe799bc 0b208896c25c03f13ed9c43b64ec6399af287871 M	dropbox
:040000 040000 e407af19578ceebfaecbbc8bca336ef1e618cc73 2d39e42d8ed6a9d6080affd1924e4b4d0b4e6fce M	generator
```

According to the documentation https://www.dropbox.com/developers/documentation/http/documentation#files-delete
This should be producing a JSON blob with an inner "metadata"

```js
{
    "metadata": {
        ".tag": "file",
        "name": "Prime_Numbers.txt",
        "id": "id:a4ayc_80_OEAAAAAAAAAXw",
[snip]
    }
}
```

However that returned above looks like this (formatted) with no inner metadata tag.

```js
{
  ".tag": "folder",
  "name": "rclone-test-lidaqez4xenaqew0lenameq6",
  "path_lower": "/rclone-test-lidaqez4xenaqew0lenameq6",
  "path_display": "/rclone-test-lidaqez4xenaqew0lenameq6",
  "id": "id:qqS0afUHwS0AAAAAAAHDTg"
}
```

That is what is causing IsMetadataFromJSON to fail

```
// IsMetadataFromJSON converts JSON to a concrete IsMetadata instance
func IsMetadataFromJSON(data []byte) (IsMetadata, error) {
	var t metadataUnion
	if err := json.Unmarshal(data, &t); err != nil {
		log.Printf("Unmarshal IsMetadataFromJSON #1: %v: %q", err, data)
		return nil, err
	}
	switch t.Tag {
	case "file":
		return t.File, nil

	case "folder":
		return t.Folder, nil

	case "deleted":
		return t.Deleted, nil

	}
	return nil, nil
}
```

The log message prints this showing the `metadata` is empty.

```
Unmarshal IsMetadataFromJSON #1: unexpected end of JSON input: ""
```

The ultimate cause seems to be the same as #47 - the SDK is calling the v1 delete rather than the v2 delete.

```diff
diff --git a/dropbox/files/client.go b/dropbox/files/client.go
index 55c53fe..c5e471f 100644
--- a/dropbox/files/client.go
+++ b/dropbox/files/client.go
@@ -1364,7 +1367,7 @@ func (dbx *apiImpl) DeleteV2(arg *DeleteArg) (res *DeleteResult, err error) {
                headers["Dropbox-API-Select-User"] = dbx.Config.AsMemberID
        }
 
-       req, err := (*dropbox.Context)(dbx).NewRequest("api", "rpc", true, "files", "delete", headers, bytes.NewReader(b))
+       req, err := (*dropbox.Context)(dbx).NewRequest("api", "rpc", true, "files", "delete_v2", headers, bytes.NewReader(b))
        if err != nil {
                return
        }
```

I will send a PR for this in just a moment :-)

**Comment by diwakergupta** [comment]
@ncw can you try with v5.4.0? Thanks again for the bug report and the detailed investigation!

**Comment by ncw** [comment]
@diwakergupta I can confirm that the problem is fixed with 5.4.0 - thanks :-)
